87.md

```diff
@@ -0,0 +1,149 @@
+NIP-87
+=======
+
+Closed Groups
+------------------
+
+Some online communities may want a measure of privacy. While nostr's architecture isn't conducive to high levels of privacy due to metadata leakage and the difficulty of implementing ratchets for key rotation, it can provide basic encrypted channels appropriate for use with large groups with dynamic member lists.
+
+# Protocol Description
+
+## Group Definition
+
+Groups are identified by the address of a replaceable `kind 35834` event. This may be wrapped using the shared key to support privately-defined groups, or published normally to support a public listing for the private group.
+
+```json
+{
+  "kind": 35834,
+  "content": "",
+  "tags": [
+    ["d", "<uuid>"],
+    ["name", "<Group name>"],
+    ["about", "<Group description>"],
+    ["picture", "<Group thumnail>"],
+    ["banner", "<Group banner image>"],
+    ["relay", "<url>"],
+    ["relay", "<url>"]
+  ]
+}
+```
+
+## Admin Key
+
+The key used to publish the group's `kind 35834` definition event MUST be used for performing privileged actions, such as publishing key rotations. Non-administrative events MAY be published to the group by the admin.
+
+Because the admin key is used for managing key rotation, it's recommended that a dedicated key be used for group administration instead of the group creator's personal key.
+
+## Key Sharing
+
+Keys are shared via a `kind 24` rumor sealed by the `admin` key, wrapped using an ephemeral key, and addressed to each group member individually, as well as to the admin's pubkey.
```

Comment:
how do we wrap keys with ephemeral keys? What is meant by an ephemeral key? I'm new to cryptography, anything I could read up on?  might be good to expand upon this on the nip?

Reply:
but wouldn't the user need the corresponding pubkey to decrypt the `privkey` tag? is it shared out of band? also how does the user get notified about being added on the group? is the `kind:24` event shared out of band as well?

Reply:
It's all wrapped inside a 1059, which is p-tagged to the recipient. So clients have to decrypt those to find out what is inside. They can decrypt them, because the payload is encrypted for the recipient's private key.

Reply:
I think @staab intended to have two kind 24 events, one with `admin` role with the editing key and one with the `member` role (terrible name) for the view-only key.

Reply:
okay I think I finally got it, hadn't gone through `https://github.com/nostr-protocol/nips/blob/master/59.md` It'd be nice to mention nip-59 on this doc. Was not aware of it and was very confused :P